File: mcorg.backends/permission-service/src/main/java/app/mcorg/permission/infrastructure/service/entities/MinecraftRequest.java
package app.mcorg.permission.infrastructure.service.entities;

public record MinecraftRequest(String identityToken) {
    public static MinecraftRequest create(String userHash, String xstsToken) {
        return new MinecraftRequest(String.format("XBL3.0 x=%s;%s", userHash, xstsToken));
    }
}

File: webapp/src/main/kotlin/app/mcorg/pipeline/auth/domain/MinecraftRequest.kt
package app.mcorg.pipeline.auth.domain

import kotlinx.serialization.Serializable

@Serializable
data class MinecraftRequest(
    val identityToken: String
)

fun createMinecraftRequest(userHash: String, xstsToken: String) = "XBL3.0 x=$userHash;$xstsToken"